main.go

```diff
@@ -1700,12 +1701,26 @@ func (tf *transformer) removeUnnecessaryImports(file *ast.File) {
 			return true
 		}
 
-		if pkg := uses.Pkg(); pkg != nil {
-			usedImports[pkg.Path()] = true
+		pkg := uses.Pkg()
+		if pkg == nil {
+			return true
 		}
 
+		var pkgName string
+		switch parent := cursor.Parent().(type) {
+		case *ast.SelectorExpr:
+			pkgNameIdent, ok := parent.X.(*ast.Ident)
+			if !ok {
+				return true // TODO(pagran): review this
+			}
+			pkgName = pkgNameIdent.Name
+		default:
+			pkgName = "."
+		}
+
+		usedImports[pkg.Path()+"|"+pkgName] = true
```

Comment:
consider `name packagePath`, with a space, because that looks more like the import declarations.